backend/src/main/java/com/huellapositiva/application/controller/ProposalApiController.java

```diff
@@ -96,7 +97,7 @@ public void createProposal(@RequestPart("dto") MultipartFile dtoMultipart,
                                @AuthenticationPrincipal String contactPersonEmail,
                                HttpServletResponse res) throws IOException {
         ProposalRequestDto dto = objectMapper.readValue(dtoMultipart.getBytes(), ProposalRequestDto.class);
-        dto.setStatus(PUBLISHED.getId());
```

Comment:
If the status cannot be set by the user, I would remove the status field from the DTO class (request specification)

Reply:
Hi Ibai, you are right. The status is never set by the user. But, depending on the person who creates the proposal, the status is different.
If the “Reviser” creates it, then, the status is directly “PUBLISHED”, if it is the “ContactPerson”, the status is “REVIEW_PENDING” instead.

So, we have different options.

1st option:
If we remove that attribute from the dto object class, still we need to set that value before saving that info into the DB., which can be done in the action class. But, in order to do that (as goes first to the parser to create the object “proposal” which will be the one we will save in the DB), then, we’ll need to add that argument to the parser, which means, sending 3 parameters to it.

2nd option:
If we don’t remove it, even though is actually not part of a dto, we don’t need to do all that from the 1st option.

What do you think? Better remove it from the dto (knowing that that will mean passing more parameters to the parser) or leave it as it is (knowing, that is not actually part of the dto)??

Thank youuu

PS: We will use that “STATUS” soon in other similar situations when the proposal needs to be edited, for example. But that part is not implemented yet.

Reply:
The request DTO should reflect the actual contract of the API, so you should not have an status field and therefore should not deserialise it with the object mapper. I guess that is aligned with your first option.
The internal implementation is up to you, the following are just available options for the action's public contract with the controller.
1- you can define two arguments in the execute method of the action, ProposalRequestDto dto, ProposalStatus status
2- you can create two public methods in the action executeByVolunteer(ProposalRequestDto) and executeByAdmin(ProposalRequestDto dto)
3- you can create two actions that extend an abstract action or implement an abstract action interface

Reply:
Hi Ibai, we think we've got it.
- We removed from the DTO the status (which makes sense). 
- Renamed the "execute" methods, to "executeByContactPerson" and "executeByAdmin".
- Set the status in the action class right after parsing the DTO into a Proposal object. (Not sure if it is the best option) Could you confirm?

Thank you